core/src/main/java/org/apache/iceberg/variants/ValueArray.java

```diff
@@ -127,4 +127,9 @@ private int writeTo(ByteBuffer buffer, int offset) {
       return (dataOffset - offset) + dataSize;
     }
   }
+
+  @Override
+  public String toString() {
```

Comment:
Minor: this probably belongs in its own PR, rather than in the writer PR.

Reply:
I need this to debug while working on this PR. So think it's useful to keep it.